site/app/views/AutoGradingView.php

```diff
@@ -0,0 +1,310 @@
+            }
+            $command = htmlentities($testcase->getDetails());
+            $testcase_message = "";
+            if ((!$testcase->isHidden() || $show_hidden) && $testcase->viewTestcaseMessage()) {
+                $testcase_message = <<<HTML
+        <span class='italics'><font color="#af0000">{$testcase->getTestcaseMessage()}</font></span>
+HTML;
+            }
+            $return .= <<<HTML
+            <h4>{$hidden_title}{$name}&nbsp;&nbsp;&nbsp;<code>{$command}</code>&nbsp;&nbsp;{$extra_credit}&nbsp;&nbsp;{$testcase_message}</h4>
+    </div>
+HTML;
+            if ($testcase->hasDetails() && (!$testcase->isHidden() || $show_hidden)) {
+                $return .= <<<HTML
+    <div id="testcase_{$count}" style="display: {$display_box};">
+HTML;
+                $autocheck_cnt = 0;
+                $autocheck_len = count($testcase->getAutochecks());
+                foreach ($testcase->getAutochecks() as $autocheck) {
+                    $description = $autocheck->getDescription();
+                    $diff_viewer = $autocheck->getDiffViewer();
+                    $return .= <<<HTML
+        <div class="box-block">
+        <!-- Readded css here so the popups have the css -->
+HTML;
+                    $title = "";
+                    $return .= <<<HTML
+            <div class='diff-element'>
+HTML;
+                    if ($diff_viewer->hasDisplayExpected()) {
+                        $title = "Student ";
+                    }
+                    if ($diff_viewer->hasDisplayActual()) {
+                        $visible = "visible";
+                        $tmp_array_string = explode("\n",trim(html_entity_decode(strip_tags($diff_viewer->getDisplayActual())), "\xC2\xA0\t")); 
+                        $less_than_30 = true;
+                        $arr_count = count($tmp_array_string);
+                        for ($x = 0; $x < $arr_count; $x++) {
+                            if(strlen($tmp_array_string[$x]) > 30) {
+                                $less_than_30 = false;
+                                $x = $arr_count;
+                            }
+                        }
+                        if (substr_count($diff_viewer->getDisplayActual(), 'line_number') < 10 && $less_than_30) {
+                            $visible = "hidden";
+                        }
+                    } else {
+                        $visible = "hidden";
+                    }
+                    $title .= $description;
+                    $return .= <<<HTML
+                <h4>{$title} <span onclick="openPopUp('{$title}', {$count}, {$autocheck_cnt}, 0)" style="visibility: {$visible}"> <i class="fa fa-window-restore" style="visibility: {$visible}; cursor: pointer;"></i></span></h4>
+                <div id="container_{$count}_{$autocheck_cnt}_0">
+                <link rel="stylesheet" type="text/css" href="{$this->core->getConfig()->getBaseUrl()}css/jquery-ui.min.css" />
```

Comment:
Well, you almost never want to import stylesheets inside the body tag of a page, but on the other hand, we don't want to spend all the time remaking the diff stuff if we don't have to (especially for large file where this operation is expensive).

I think a better solution would be add a new parameter to openPopUp `$css` which you could pass in a list of all CSS that you want in the popup an then just dynamically add them in the function.